### PR TITLE
fix: get correct link for a space in all spaces grid

### PR DESCRIPTION
**Pull request by joaoviana**
Closes: #5401 

### Description:
When viewing All Spaces, each Link has an empty string for `projectUuid`. 
This only happens on spaces. 

Before
(see issue)

After
image

### CODE REVIEW

File: packages/frontend/src/components/common/ResourceView/resourceUtils.ts [hunk 1]
```diff
@@ -50,7 +50,9 @@ export const getResourceUrl = (projectUuid: string, item: ResourceViewItem) => {
         case ResourceViewItemType.CHART:
             return `/projects/${projectUuid}/saved/${item.data.uuid}`;
         case ResourceViewItemType.SPACE:
-            return `/projects/${projectUuid}/spaces/${item.data.uuid}`;
+            return `/projects/${projectUuid || item.data.projectUuid}/spaces/${
+                item.data.uuid
+            }`;
```
Comment (joaoviana) [inline packages/frontend/src/components/common/ResourceView/resourceUtils.ts:55]: in `ResourceViewGrid` we set some props to empty strings by default:
```
  pinnedItemsProps = { projectUuid: '', pinnedListUuid: '' },
```

There could be another way of solving this bug but this is a quick fix for now. Essentially we should accommodate no empty strings for space items? What do we think?

Comment (IrakliJani) [reply packages/frontend/src/components/common/ResourceView/resourceUtils.ts:55]: @yasmin-e was this a breaking change from the backend changes? maybe it should be fixed in the backend?

Comment (ZeRego) [reply packages/frontend/src/components/common/ResourceView/resourceUtils.ts:55]: Looks like a FE issue where the projectUuid property is not being passed to this component. Usually, the project uuid is part of the url.

Comment (joaoviana) [reply packages/frontend/src/components/common/ResourceView/resourceUtils.ts]: Absolutely @ZeRego, as you've mentioned it was a props issue. 

Have done this instead, what do we think? https://github.com/lightdash/lightdash/pull/5410/commits/dd26f9fe00a07d41b24100d5b87b52bfc04fbb2a

(cc @IrakliJani , @yasmin-e )

**Pull request merged**